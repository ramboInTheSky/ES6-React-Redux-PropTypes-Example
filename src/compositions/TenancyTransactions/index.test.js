import React from 'react';
import { shallow } from 'enzyme';

import { TenancyTransactions } from './';

describe('<TenancyTransactions />', () => {
  let defaultProps;
  let component;

  beforeEach(() => {
    defaultProps = {
      customerAccounts: [
        {
          paymentReference: '313131',
          typeCode: 'REN',
          typeName: 'RENT ACCOUNT',
          transactions: [
            {
              amount: 173.59,
              balanceAtEffectiveDate: 1388.72,
              charge: 'DEBIT',
              creditValue: 0,
              debitValue: 173.59,
              description: 'Weekly Charge',
              effectiveDate: '2018-05-28T00:00:00',
              subType: '',
              type: 'DRS',
            },
          ],
        },
      ],
      transactionHistoryText: {
        currencySymbol: 'Â£',
        heading: 'Transaction history (12 months)',
        effectiveDate: 'Date',
        description: 'Description',
        paidIn: 'Paid in',
        amount: 'Amount',
        charged: 'Charged',
        balance: 'Balance',
      },
    };

    component = shallow(<TenancyTransactions {...defaultProps} />);
  });

  it('renders correctly with full props', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders null with no customerAccounts', () => {
    const props = {
      customerAccounts: [],
    };
    const wrapper = shallow(<TenancyTransactions {...defaultProps} {...props} />);
    expect(wrapper.get(0)).toBe(null);
  });

  it('renders null with customerAccount but with no transactions', () => {
    const props = {
      customerAccounts: [
        {
          transactions: [],
        },
      ],
    };
    const wrapper = shallow(<TenancyTransactions {...defaultProps} {...props} />);
    expect(wrapper.get(0)).toBe(null);
  });
});
