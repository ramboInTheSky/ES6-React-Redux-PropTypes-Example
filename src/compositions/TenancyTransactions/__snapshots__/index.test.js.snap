// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TenancyTransactions /> renders correctly with full props 1`] = `
<React.Fragment>
  <Styled(styled.h3)>
    Transaction history (12 months)
  </Styled(styled.h3)>
  <styled.div
    data-bdd="transactionHistory-RENundefined"
    key="transactionHistory-RENundefined"
  >
    <Styled(styled.p)>
      RENT ACCOUNT
    </Styled(styled.p)>
    <TransactionHistoryList
      dataBdd="transactionHistoryList"
      text={
        Object {
          "amount": "Amount",
          "balance": "Balance",
          "charged": "Charged",
          "currencySymbol": "Â£",
          "description": "Description",
          "effectiveDate": "Date",
          "heading": "Transaction history (12 months)",
          "paidIn": "Paid in",
        }
      }
      transactions={
        Array [
          Object {
            "amount": 173.59,
            "balanceAtEffectiveDate": 1388.72,
            "charge": "DEBIT",
            "creditValue": 0,
            "debitValue": 173.59,
            "description": "Weekly Charge",
            "effectiveDate": "2018-05-28T00:00:00",
            "subType": "",
            "type": "DRS",
          },
        ]
      }
    />
  </styled.div>
</React.Fragment>
`;
